src/agent/src/register.cpp

```diff
@@ -1,42 +1,52 @@
 #include <register.hpp>
 
-#include <agent_info.hpp>
-#include <configuration_parser.hpp>
+#include <http_client.hpp>
 #include <logger.hpp>
 
 #include <boost/beast/http.hpp>
 #include <nlohmann/json.hpp>
 
 namespace http = boost::beast::http;
 
-namespace registration
+namespace agent_registration
 {
-    bool RegisterAgent(const UserCredentials& userCredentials, http_client::IHttpClient& httpClient)
+    AgentRegistration::AgentRegistration(std::string user,
+                                         std::string password,
+                                         const std::string& key,
+                                         const std::string& name)
+        : m_managerIp(m_configurationParser.GetConfig<std::string>("agent", "manager_ip"))
+        , m_managerPort(m_configurationParser.GetConfig<std::string>("agent", "server_mgmt_api_port"))
+        , m_user(std::move(user))
+        , m_password(std::move(password))
     {
-        const configuration::ConfigurationParser configurationParser;
-        const auto managerIp = configurationParser.GetConfig<std::string>("agent", "manager_ip");
-        const auto managerPort = configurationParser.GetConfig<std::string>("agent", "server_mgmt_api_port");
 
-        const auto token = httpClient.AuthenticateWithUserPassword(
-            managerIp, managerPort, userCredentials.user, userCredentials.password);
+        m_agentInfo.SetKey(key);
+        if (!name.empty())
```

Comment:
You are missing the latest changes here that set the name to the hostname when not specified, please fix this.

Reply:
Done